listenbrainz/webserver/static/js/src/LastFMImporter.test.tsx

```diff
@@ -368,6 +369,75 @@ describe("submitPage", () => {
   });
 });
 
+describe("getUserPrivacy", () => {
+  beforeEach(() => {
+    const wrapper = shallow<LastFmImporter>(<LastFmImporter {...props} />);
+    instance = wrapper.instance();
+    instance.setState({ lastfmUsername: "dummyUser" });
+
+    // Needed for startImport
+    instance.APIService.getLatestImport = jest.fn().mockImplementation(() => {
+      return Promise.resolve(0);
+    });
+    window.fetch = jest.fn().mockImplementationOnce(() => {
+      return Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve(getInfo),
+      });
+    });
+  });
+
+  it("should call with the correct url", () => {
+    instance.getUserPrivacy();
+
+    expect(window.fetch).toHaveBeenCalledWith(
+      `${props.lastfmApiUrl}?method=user.getrecenttracks&user=${instance.state.lastfmUsername}&api_key=${props.lastfmApiKey}&format=json`
+    );
+  });
+
+  it("should return user privacy status", async () => {
+    // mock function for fetch (no data.error)
+    window.fetch = jest.fn().mockImplementationOnce(() => {
+      return Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve(page),
+      });
+    });
+    expect(await instance.getUserPrivacy()).toEqual(false);
```

Comment:
While this does work as expected, I think it may be wise to use only one format in these tests (`expect( await…).toEqual` vs. `await expect(…).resolves.toEqual`)

I suppose `await expect(x).resolves.to…` is more flexible and ensures promise rejections are caught cleanly (and allows us to test `.rejects.toThrowError()`  as you did below), so let's go with that format.